src/storage/assets/assetsActions.ts

```diff
@@ -41,3 +43,19 @@ export const syncNetworkTokensInfo = createAsyncThunk('assets/syncNetworkTokensI
 
   return metadata
 })
+
+export const syncUnknownTokensInfo = createAsyncThunk(
+  'assets/syncUnknownTokensInfo',
+  async (unknownTokenIds: Asset['id'][]) => {
+    const tokens = await Promise.all(
```

Comment:
One thing to keep in mind here, that caused me troubles on the explorer: `Promise.all` is all or nothing. If any of the promises in the map fails, the whole sync will fail. This can be problematic if this ends up in showing incomplete lists of assets.

Reply:
Thanks for bringing my attention to this!

After reading some doc pages and articles, I think that for this particular case, this is the best way to do it:

```ts
    const tokens = await Promise.allSettled(
      unknownTokenIds.map((id) =>
        client.node.fetchStdTokenMetaData(id).then((data) => ({
          id,
          ...data
        }))
      )
    )

    return (tokens.filter((result) => result.status === 'fulfilled') as PromiseFulfilledResult<TokenMetaData>[]).map(
      ({ value: { totalSupply, ...rest } }) => rest
    )
```

1. `Promise.all` is not optimal because the data returned from each promise is not interdependent, so we don't want to fail the whole thing if any of them fails. Even though `Promise.all` is the most performant solution, we want more granular control, so that we can "save" the promises that fulfill and discard the ones that reject.
2. `for await...of` would be a good solution if we wanted to handle each rejection separately. For this particular case, I don't think we care to react when one of these promises fails. We consider the token as `unknown` directly.
3. `Promise.allSettled` is the best solution in my opinion. It returns an array with a `status` for each item to inform whether it fulfilled or rejected. If it's fulfilled, it includes the return value. It's faster than `for await...of` and it gives more granular control than `Promise.all`.